app/src/main/java/com/spascoding/taskycourse/feature_register_screen/domain/repository/AuthRepository.kt

```diff
@@ -4,15 +4,8 @@ import com.spascoding.taskycourse.feature_register_screen.data.remote.model.Logi
 import retrofit2.Response
 
 interface AuthRepository {
-    suspend fun register(
-        name: String,
-        email: String,
-        password: String,
-    ): Response<Void>
-    suspend fun login(
-        email: String,
-        password: String,
-    ): Response<LoginResponse>
-    fun authenticate()
+    suspend fun register(name: String, email: String, password: String): Response<Void>
+    suspend fun login(email: String, password: String): Response<LoginResponse>
```

Comment:
By referencing `Response` and `LoginResponse` here, you're coupling your domain layer to data layer classes. Processing such HTTP responses is job of the repository and not the ViewModel. I recommend introducing a generic result wrapper class you can use for any kind of result across the app.

I personally love this version:
```
sealed interface Result<out D, out E: Error> {
    data class Success<out D>(val data: D): Result<D, Nothing>
    data class Error<out E: com.plcoding.tasky.core.domain.util.Error>(val error: E): Result<Nothing, E>
}

inline fun <T, E: Error, R> Result<T, E>.map(map: (T) -> R): Result<R, E> {
    return when(this) {
        is Result.Error -> Result.Error(error)
        is Result.Success -> Result.Success(map(data))
    }
}

fun <T, E: Error> Result<T, E>.asEmptyDataResult(): EmptyResult<E> {
    return map {  }
}

inline fun <T, E: Error> Result<T, E>.onSuccess(action: (T) -> Unit): Result<T, E> {
    return when(this) {
        is Result.Error -> this
        is Result.Success -> {
            action(data)
            this
        }
    }
}
inline fun <T, E: Error> Result<T, E>.onError(action: (E) -> Unit): Result<T, E> {
    return when(this) {
        is Result.Error -> {
            action(error)
            this
        }
        is Result.Success -> this
    }
}

typealias EmptyResult<E> = Result<Unit, E>
```
It handles data and errors in a type-safe way. You'll need a marker interface to mark error enums:
```
interface Error
```
And then you can extend this across the codebase with different error types, e.g.:
```
sealed interface DataError: Error {
    enum class Remote: DataError {
        NOT_FOUND,
        NO_INTERNET,
        UNAUTHORIZED,
        ...
    }
    
    enum class Local: DataError {
        DISK_FULL,
        ...
    }
}
```
Works great:
```
repository.login(payload).onSuccess { user ->

}.onError { error -> 
    // toUiText() can be an extension of DataError which maps an error to the corresponding message
    // using my UiText wrapper (https://youtu.be/mB1Lej0aDus?si=wBpL7IezHVMc2ZwT)
    _state.update { it.copy(error = error.toUiText()) }
}
```